src/consensus/encode.rs

```diff
@@ -597,6 +644,33 @@ pub(crate) fn consensus_encode_with_size<S: io::Write>(data: &[u8], mut s: S) ->
 }
 
 
+struct ReadBytesFromFiniteReaderOpts {
+    len: usize,
+    chunk_size: usize,
+}
+
+/// Read `len` bytes from reader, where `len` could potentially be malicious
```

Comment:
This comment is stale, there is no `len` argument now. This PR was a little confusing to review because the last patch modifies code in the first patch.